File: Leetcode/4216-weighted-word-mapping/solution.cpp
class Solution {
public:
    string mapWordWeights(vector<string>& words, vector<int>& weights) {
        string answer = "";
        for(auto &word:words){
            int result = 0;
            for(auto &c:word)
                result += weights[c-'a'];
            answer += char(25 - (result)%26+'a');
        }
        return answer;
    }
};

File: Leetcode/4216-weighted-word-mapping/solution.java
class Solution {
    public String mapWordWeights(String[] words, int[] weights) {
        String ans = "";
        for(String word:words){
            int weight = 0;
            for(int i=0;i<word.length();i++)
                weight += weights[word.charAt(i)-'a'];
            weight = weight%26;
            ans += (char)('a'+25-weight);
        }
        return ans;
    }
}
